Update RELEASE NOTES (breaking changes), README (link to release notes) and DOCUMENTATION (#608)
giraffe-fsharp/Giraffe: RELEASE_NOTES.md

## 7.0.0-alpha-001 - 2024-07-08

> #### Breaking changes

Reviewer: Perfect, make it stand out! :)

> - `NewtonsoftJson` and `SystemTextJson` modules were removed by PR [Promote System.Text.Json as default JSON serializer](https://github.com/giraffe-fsharp/Giraffe/pull/563).

Reviewer: Perhaps:

```suggestion
- `NewtonsoftJson` and `SystemTextJson` modules are removed, the latter is now module `Json`, NewtonSoft is not supported anymore, see PR [Promote System.Text.Json as default JSON serializer](https://github.com/giraffe-fsharp/Giraffe/pull/563).
```

>     - Check the updated JSON documentation section to know more: [link](https://github.com/giraffe-fsharp/Giraffe/blob/master/DOCUMENTATION.md#json). Notice that there we have instructions on how to setup `NewtonsoftJson` back if you need.

Reviewer: Maybe this?

```suggestion
    - For more information, check the updated JSON documentation section: [link](https://github.com/giraffe-fsharp/Giraffe/blob/master/DOCUMENTATION.md#json). This has instructions on how to setup `NewtonsoftJson` back if you need to keep it for your project.
```
